zipline/finance/performance/position.py

```diff
@@ -65,22 +65,24 @@ def earn_dividend(self, dividend):
         Register the number of shares we held at this dividend's ex date so
         that we can pay out the correct amount on the dividend's pay date.
         """
-        assert dividend['sid'] == self.sid
-        out = {'id': dividend['id']}
+        out = {}
+
+        out['amount'] = self.amount * dividend.amount
+        return out
+
+    def earn_stock_dividend(self, stock_dividend):
+        """
+        Register the number of shares we held at this dividend's ex date so
```

Comment:
I think this comment (and the one for `earn_dividend` above) isn't accurate anymore, right?

`earn_dividend` is probably something like "Calculates the and returns the dollar amount of the dividend based on current position."

same for `earn_stock_dividend`